### PR TITLE
Mark a test as "Not Run" if no data is collected

### LINKED ISSUE #150: Running a test without collecting any data registers as a "runned test" in the logger
Currently, you can start a test and then end it without enabling the robot and the logger doesn't keep the test at "Not Run".

### PR DESCRIPTION
**Pull request by Piphi5**
Fixes #150

### CODE REVIEW

File: sysid-application/src/main/native/cpp/telemetry/TelemetryManager.cpp [hunk 1]
```diff
@@ -141,8 +140,16 @@ void TelemetryManager::EndTest() {
             "\nNOTE: the robot stopped recording data early because the entry "
             "storage was exceeded.";
       }
-      func(msg);
+
+    } else {
+      msg = "No data was detected.";
     }
+    func(msg);
```
Comment (calcmogul) [inline sysid-application/src/main/native/cpp/telemetry/TelemetryManager.cpp:147]: What does func() do?

Comment (Piphi5) [reply sysid-application/src/main/native/cpp/telemetry/TelemetryManager.cpp:147]: It's a function that displays any telemetry messages onto the GUI

**Comment by calcmogul** [comment]
I'm not sure how to test this PR.

**Comment by Piphi5** [comment]
I tested it by connecting to the analysis-test project and then starting and canceling the quasistatic test and making sure that the test was set to `Not Run` after it was prematurely canceled.

**Review by calcmogul — APPROVED** [review]
I confirmed main has the issue and this PR fixes it.

**Pull request merged**